ma_search/data/shape.py

```diff
@@ -17,52 +17,74 @@
 limitations under the License.
 """
 
-import uuid
+import json
 import logging
+from pathlib import Path
+from uuid import UUID, uuid4
 
 import ma_search
+from ma_search.common import logException
+from shapely.geometry import MultiPolygon, Polygon, mapping, shape
 
 logger = logging.getLogger(__name__)
 
-
-class Shape():  # pragma: no cover
-
+class Shape():
     def __init__(self, uuid):
+        """Keeps track, can lookup and manipulate (Multi)Polygons via their uuid
 
+        Args:
+            uuid (str): Unique identifyer UUID4
+        """
         self.conf = ma_search.CONFIG
         self._uuid = uuid
+        if not self._validateUuid():
+            logger.error("UUID %s is not valid", self._uuid)
+            return
 
-        # ToDo:
-        # - Validate UUID
-        # - Check that file exists
-
-        return
+        self._path = (Path(self.conf.dataPath) / self._uuid).with_suffix(".geojson")
+        if not self._path.exists():
+            logger.error("UUID file %s does not exist", self._path)
 
     @classmethod
     def fromGeoJSON(cls, data):
-        """
+        """Returns Shape instance from a GeoJson dict or file
+
+        Parameters
+        ----------
+        data : dict or Path or str
+            Input data, either a single GeoJson feature, or only the geometry
+            section of a single feature. If str or Path: Path to the file
+            containing this dict.
+
         Returns
         -------
-        :obj:`Shape` :
-            A new instance of Polygon()
+        :obj:`Shape` or None:
+            A new instance of Shape(). Returns None for invalid input, e.g. not
+            a Polygon.
         """
-        newUUID = uuid.uuid4()
-        # save geojson data to file named newUUID.geojson
-        poly = cls(newUUID)
+        uuid = uuid4()
+        if cls.polygonFromGeoJson(data) is None:
+            return None # tests that it
+
+        path = (ma_search.CONFIG.dataPath / str(uuid)).with_suffix(".geojson")
+        with path.open('w') as f:
```

Comment:
All file system operations should be in try/except. Wrong quote marks around "w".